backends/ebpf/runtime/ebpf_test.h

```diff
@@ -99,16 +99,14 @@ typedef unsigned long long u64;
 # error "Fix your compiler's __BYTE_ORDER__?!"
 #endif
 
-#define load_byte(data, b)  (*(((u8*)(data)) + (b)))
-#define load_half(data, b) __constant_ntohs(*(u16 *)((u8*)(data) + (b)))
-#define load_word(data, b) __constant_ntohl(*(u32 *)((u8*)(data) + (b)))
-static u64 load_dword(void *skb, u64 off) {
-  return ((u64)load_word(skb, off) << 32) | load_word(skb, off + 4);
-}
-
 #define htonl(d) __constant_htonl(d)
 #define htons(d) __constant_htons(d)
 
+#define load_byte(data, b)  (*(((u8*)(data)) + (b)))
```

Comment:
Hmm, I think just interpreting the bytes as is makes the most sense. Would it be sensible to stop using int and instead operate on byte arrays of various sizes? The cast to int is what is causing trouble when trying to interpret the bytes, at least with the 48 bit arrays. Not using any order conversion works fine for eBPF and userspace testing, they are equivalent.
Unfortunately, that is not the case for XDP, I do not exactly remember the cause of this behavior. In the XDP tests, I have not tested using XDP tables yet. 

Regarding the single file, I could create a base file from which ebpf_test, ebpf_kernel, and ebpf_xdp inherit from. But this would make things like adding new targets quite messy, wouldn't it?

Reply:
There is no reason to have this code duplicated.
But you still have changed the meaning: the old code used to have host order, now you have network order.
This means that any arithmetic on the header fields will produce wrong results on a machine where the host order is not the network order. I don't know if we have any tests using arithmetic.

Reply:
So, yes, you should move this code to a common header file. I don't see how adding a new target is impacted by this code being in a separate file; this code will never change for any target.

Reply:
I reverted back to the network byte order conversion. Still, I think operating on the standard byte order directly makes more sense and is less confusing. We could have conversion for arithmetic only? Not sure how that would work with the compiler.
I do not think we have arithmetic tests unfortunately. I tried to emulate the arith-bmv2 tests and implement one, but was unable to get it to work, I am unsure which operations are available in the ebpf framework.

Reply:
We have issue #1122 to track this problem, but let's fix that separately.